File: app/src/main/java/com/test/xyz/demo/presentation/common/DisposableManager.java
package com.test.xyz.demo.presentation.common;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class DisposableManager {
    private CompositeDisposable compositeDisposable;

    public void add(Disposable disposable) {
        if (disposable != null) {
            getCompositeDisposable().add(disposable);
        }
    }

    public void dispose() {
        if (compositeDisposable != null) {
            compositeDisposable.dispose();
        }
    }

    private CompositeDisposable getCompositeDisposable() {
        if (compositeDisposable == null || compositeDisposable.isDisposed()) {
            compositeDisposable = new CompositeDisposable();
        }

        return compositeDisposable;
    }
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/common/DisposableManager.kt
package com.test.xyz.demo.presentation.common

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

class DisposableManager {
    private var compositeDisposable: CompositeDisposable? = null

    fun add(disposable: Disposable?) {
        if (disposable != null) {
            getCompositeDisposable()?.add(disposable)
        }
    }

    fun dispose() {
        compositeDisposable?.dispose()
    }

    private fun getCompositeDisposable(): CompositeDisposable? {
        if (compositeDisposable == null || compositeDisposable!!.isDisposed) {
            compositeDisposable = CompositeDisposable()
        }

        return compositeDisposable
    }
}
